File: misc/how_many_xs.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution {
  int count(int n, int x) {
    int ans = 0;
    while (n > 0) {
      if (n % 10 == x)
        ++ans;
      n /= 10;
    }
    return ans;
  }

public:
  int countX(int L, int R, int X) {
    // code here
    int ans = 0;
    for (int i = L + 1; i < R; ++i) {
      ans += count(i, X);
    }
    return ans;
  }
};

//{ Driver Code Starts.
int main() {
  int t;
  cin >> t;
  while (t--) {
    int L, R, X;
    cin >> L >> R >> X;
    Solution ob;
    int ans = ob.countX(L, R, X);
    cout << ans << "\n";
  }
}

// } Driver Code Ends

File: misc/how_many_xs.py
# User function Template for python3


class Solution:
    def countX(self, L, R, X):
        # code here
        def f(a):
            ans = 0
            while a > 0:
                if a % 10 == X:
                    ans += 1
                a //= 10
            return ans

        ans = 0
        for i in range(L + 1, R):
            ans += f(i)

        return ans


# {
# Driver Code Starts
# Initial Template for Python 3

import math

if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        L, R, X = map(int, input().strip().split(" "))
        ob = Solution()
        ans = ob.countX(L, R, X)
        print(ans)
# } Driver Code Ends
